core/node/da_clients/src/avail/sdk.rs

```diff
@@ -450,13 +450,24 @@ impl GasRelayClient {
             .post(&submit_url)
             .body(Bytes::from(data))
             .header("Content-Type", "text/plain")
-            .header("Authorization", &self.api_key)
+            .header("Authorization", format!("Bearer {}", self.api_key))
             .send()
-            .await?;
+            .await
+            .map_err(|e| anyhow::anyhow!("Failed to submit data to the gas relay: {:?}", e))?;
```

Comment:
Nit: Is there anything wrong with using `context()` here? It would result in a more structured error.